src/event-handler/index.ts

```diff
@@ -0,0 +1,85 @@
+import { EventEmitter } from 'events';
+import Ain from '../ain';
+import {
+  BlockchainEventTypes,
+  BlockFinalizedEventConfig,
+  EventConfigType,
+  ValueChangedEventConfig,
+} from '../types';
+import EventFilter from './event-filter';
+import EventChannelManager from './event-channel-manager';
+
+export default class EventHandler {
+  private readonly _filters: { [filterId: string]: EventFilter };
+  private readonly _eventEmitters: { [filterId: string]: EventEmitter };
+  private _ain: Ain;
+  private _eventChannelManager: EventChannelManager;
+
+  constructor(ain: Ain) {
+    this._filters = {};
+    this._eventEmitters = {};
+    this._ain = ain;
+    this._eventChannelManager = new EventChannelManager(ain, this);
+  }
+
+  async connect() {
+    await this._eventChannelManager.connect();
+  }
+
+  disconnect() {
+    this._eventChannelManager.disconnect();
+  }
+
+  emitEvent(filterId: string, payload: any) {
+    const targetEventEmitter = this._eventEmitters[filterId];
+    if (!targetEventEmitter) {
+      throw Error(`Can't find event emitter by filter id (${filterId})`);
+    }
+    targetEventEmitter.emit('data', payload);
+  }
+
+  emitError(filterId: string, errorMessage: string) {
```

Comment:
what about extracting these two functions (emitEvent, emitError) to a separate module e.g. EventCallbackManager()?

Reply:
Very cool structure. I moved these functions to EventCallbackManager. Thanks 👍